maxline-2.md

```diff
@@ -0,0 +1,112 @@
+---
+title: oragono.io/maxline-2 Capability
+layout: spec
+meta-description: A method for negotiating longer IRC protocol lines.
+copyrights:
+  -
+    name: "Daniel Oaks"
+    period: "2016-2019"
+    email: "daniel@danieloaks.net"
+---
+## Introduction
+There has been a desire to allow longer lines while sending or receiving IRC traffic, allowing longer messages with `PRIVMSG`/`NOTICE` and doing away with (or at least working around) the original limit of 512 bytes. However, doing so in a backwards-compatible way requires opt-in from clients and explicit action from servers.
+
+The `oragono.io/maxline-2` capability specifies the maximum length of the 'rest' of the message (everything that isn't the tags section), that the server allows. Lines are truncated or split to account for clients which have not negotiated this extended length.
+
+
+## Architecture
+
+### The Capability
+The `oragono.io/maxline-2` capability, when advertised, MUST have a value which is a positive integer.
+
+For example:
+
+    C: CAP LS 302
+    S: CAP * LS :oragono.io/maxline-2=2048
+
+To match standard message handling, tags and the rest of the message have separate length values. The value of this capability represents the maximum number of bytes that the 'rest' section of the message can take up.
+
+As an example, if `maxline` is `1024` then the maximum size of a full IRC message would be `<taglength>+1024` bytes (the standard [length of the tags section](https://ircv3.net/specs/extensions/message-tags.html#size-limit) plus 1024 for the rest of the message).
+
+Servers MUST truncate incoming messages to the `maxline` value before processing them.
+
+The `maxline` value MUST be at least `512` and SHOULD default to at least `2048`.
+
+### Interaction with PRIVMSG and NOTICE
+If a client has negotiated the `oragono.io/maxline-2` capability and sends a `PRIVMSG` or a `NOTICE` message that is longer than 512 bytes, the receiving server MUST split this into multiple regular length messages (maximum 512 bytes) when sending it to clients that have not negotiated this capability.
+
+Servers SHOULD split on whitespace or other word boundaries, and may use whatever method is easiest for them to implement. Lines MUST NOT be split in the middle of a UTF-8 character.
+
+Servers MAY split other commands/numerics into multiple lines in whatever way ensures the best compatibility and functionality.
+
+### The `oragono.io/truncated` Tag
+The `oragono.io/truncated` tag, when present, indicates that a message has been truncated due to the client's line length. It may be sent to any client which has negotiated the [`message-tags` capability](https://ircv3.net/specs/extensions/message-tags.html#message-tags-capability).
```

Comment:
Should this "may" be a "MAY" (or a "SHOULD")?

Reply:
Hmm, given the sort of behavior this refers to, it makes sense to leave it as is I think.